app/src/main/java/com/example/tasky/di/AppModule.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.tasky.di
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.example.tasky.TaskyApp
+import com.example.tasky.common.Constants
+import com.example.tasky.util.ApiKeyInterceptor
+import com.example.tasky.util.ApiTokenInterceptor
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AppModule {
+
+    @Provides
+    @Singleton
+    fun provideSharedPrefs(app: TaskyApp): SharedPreferences {
+        return app.getSharedPreferences("Prefs", Context.MODE_PRIVATE)
+    }
+
+    @Provides
+    @Singleton
+    fun provideToken(prefs: SharedPreferences): String? {
+        return prefs.getString("jwt", null)
+    }
```

Comment:
Avoid providing things that can change over time. This is a singleton here and will never change during one app lifetime. If the user logs out and logs in again, the token will change, but the singleton won't. That'll lead to a bug. Just expose the prefs object and use it to read the token every time